third_party/move/move-vm/types/src/loaded_data/struct_name_indexing.rs

```diff
@@ -90,27 +92,49 @@ impl StructNameIndexMap {
             }
 
             let idx = index_map.backward_map.len();
-            index_map.backward_map.push(backward_value);
+            index_map.backward_map.push((backward_value, module_hash));
             index_map.forward_map.insert(forward_key, idx);
             idx
         };
 
         Ok(StructNameIndex(idx))
     }
 
+    /// Maps the struct identifier into an index. If the identifier already exists returns the
+    /// corresponding index. This function guarantees that for any struct identifiers A and B,
+    /// if A == B, they have the same indices.
+    pub fn struct_name_to_idx(
```

Comment:
Probably better to make this test only, seems like it is used for tests so someone does not accidentally use it and end up computing hashes.